ocs/ocp.py

```diff
@@ -134,23 +148,48 @@ def apply(self, yaml_file):
         command = f"apply -f {yaml_file}"
         return self.exec_oc_cmd(command)
 
-    def wait_for_resource_status(
-        self, resource_name, condition, timeout=30, sleep=3
+    def wait(
```

Comment:
@RazTamir ,

My point here was, if we already have a "oc wait" command which does same job , why are you using indirect method of achieving same result using "get" method ?
From openshift doc this is what I understood, 

```
wait
Wait for a specific condition on one or more resources.

Example: Wait for the python-1-mz2rf Pod to be deleted
$ oc wait --for=delete pod/python-1-mz2rf
```
If this is what exactly you are doing why can't you just wrap "oc wait " command instead of using "oc get" cli. Also "oc wait --timeout=yyy" is  available.

Am I missing something here ?

Reply:
Hi @shylesh ,
I think I taked about the differences from oc wait cmd in the automation session yesterday (watch it (: )
From what I saw, oc wait can be used for delete and pod/ready which is very limited to our needs.
In that implementation, you are open to any condition and any resource to sample.
Also, you can specify how many resources should be in that condition

Reply:
shouldn't we rename the function name to wait_for_resource than case to not confuse with wait form cli?

Reply:
> Hi @shylesh ,
> I think I taked about the differences from oc wait cmd in the automation session yesterday (watch it (: )
> From what I saw, oc wait can be used for delete and pod/ready which is very limited to our needs.
> In that implementation, you are open to any condition and any resource to sample.
> Also, you can specify how many resources should be in that condition

Hi @RazTamir ,

I don't think "oc wait" is specific to pod and delete ops. we can exercise it for any resource and conditions will be specific to resources. For ex: for pods condition could be Creating, Running etc whereas for deployments condition=Running doesn't hold and we have to check list of applicable conditions.

I feel finding number of resources matching your selector deserves its own function rather than combining it with wait. In general wait semantics should never be worried about count of specific condition match and only either fail based on timeout OR return True if  condition met by resources shortlisted through selectors. It should be upto the consumer to decide whether they want count OR not based on context of test and may issue further get call.

Reply:
@RazTamir 

I did a dirty cli experiment and it works may not be perfect one.
```bash
 $ ~/oc-tools/oc wait --timeout=10s  -l rook_cluster=rook-ceph deployment --for=condition=available -n  rook-ceph -o=jsonpath='{.kind}{"\n"}' | wc -l
7
```
This returns number of deployments which satisfies given condition.

Reply:
+1 on your comments.

Few things in my mind:
1) At some scenarios, the pods will spin up after a few seconds and not right away which means the return value from this cli command will be 0 because when you executed it, the number of pods is still 0
2) I will split the function into wait() and wait for_resouce_count() and hope this will serve your feedback
3) the fact that a cli command provides a functionality doesn't mean we have to stick with it.

Reply:
> +1 on your comments.
> 
> Few things in my mind:
> 
> 1. At some scenarios, the pods will spin up after a few seconds and not right away which means the return value from this cli command will be 0 because when you executed it, the number of pods is still 0

[shylesh] Not necessarily, because wait will not return until condition is met OR timeout. If pod has condition=Running that means pod is up and running so I guess there could be no window
of missing the pod, though I haven't tried.  

> 2. I will split the function into wait() and wait for_resouce_count() and hope this will serve your feedback
[shylesh] that would work.

> 3. the fact that a cli command provides a functionality doesn't mean we have to stick with it.
[Shylesh] I agree. I was just curious to see how that command works hence I tried.

Reply:
@shylesh , I'm just going through all data I can find regarding oc wait and I can't find too much information about waiting on different conditions other than available/ready.

For now, I will change the name of the function to wait_for_resource and will leave the implementation of a pure wait to later phase

Reply:
> @shylesh , I'm just going through all data I can find regarding oc wait and I can't find too much information about waiting on different conditions other than available/ready.
> 
> For now, I will change the name of the function to wait_for_resource and will leave the implementation of a pure wait to later phase

Yes, please go ahead as this is one of the useful feature required for writing tests. We can revisit this later.